src/targets/registry.ts

```diff
@@ -101,7 +98,34 @@ export class RegistryTarget extends BaseTarget {
   /**
    * Extracts Registry target options from the raw configuration.
    */
-  public getRegistryConfig(): RegistryConfig {
+  public getRegistryConfig(): RegistryConfig[] {
+    if (!this.config.sdks && !this.config.apps) {
+      this.logger.warn(
+        'You are using a deprecated registry traget config, please update.'
+      );
+      return [this.getLegacyRegistryConfig()];
+    }
+
+    const items = [];
+    for (const canonicalName of Object.keys(this.config.sdks || {})) {
+      items.push({
+        ...this.config.sdks[canonicalName],
+        type: RegistryPackageType.SDK,
+        canonicalName,
+      });
+    }
+    for (const canonicalName of Object.keys(this.config.apps || {})) {
+      items.push({
+        ...this.config.apps[canonicalName],
+        type: RegistryPackageType.APP,
+        canonicalName,
+      });
+    }
+
+    return items;
```

Comment:
How is it that we can now skip all of the validation that's now in `getLegacyRegistryConfig`? I don't see that we added type-checking that wasn't already there, e.g.

Reply:
Is the answer related to this in the PR description?

> This new structure also has the side effect of making it very hard to
miss required configuration.

Reply:
>Is the answer related to this in the PR description?

Yes! We should still type-check this but I'm hoping to do that in a future diff where I enable strict target config type checks through `validateConfig` in `config.ts`.